src/main/java/seedu/duke/template/BinaryTree.java

```diff
@@ -11,17 +13,38 @@ public BinaryTree(T root) {
         arrayList.set(0, root);
     }
 
+    public BinaryTree() {
+        this(null);
+    }
+
+    /**
+     * Checks if the tree is empty.
+     *
+     * @return boolean value, true if the tree is empty, otherwise false.
+     */
     public boolean isEmpty() {
         return arrayList.isEmpty();
     }
 
+    /**
+     * Returns the root of the tree.
+     *
+     * @return T type value, root of the tree.
+     */
     public T getRoot() {
```

Comment:
Should this check for `isEmpty()` first?

Reply:
Actually, the list is initally populated with 16 null values, so technically we gotta check for if the 1st value is null. Might have to change the isEmpty() function.